cardano-testnet/src/Testnet/Cardano.hs

```diff
@@ -190,8 +195,15 @@ mkTopologyConfig numNodes allPorts port True = J.encode topologyP2P
         []
         (P2P.UseLedger DontUseLedger)
 
-cardanoTestnet :: CardanoTestnetOptions -> H.Conf -> H.Integration TestnetRuntime
-cardanoTestnet testnetOptions configuration@H.Conf {tempAbsPath, testnetMagic} = do
+data RunTest
+  = StartTestnetOnly
+  | TestShutdownOnSlotSynced
+  | TestStakeSnapshot
+  | TestLeadershipSchedule
+  deriving (Show)
+
+cardanoTestnet :: RunTest -> CardanoTestnetOptions -> H.Conf -> H.Integration TestnetRuntime
```

Comment:
The vision I have for this function it to simply be concerned with starting a testnet not specific tests. We're heading in the wrong direction here.